### PR TITLE
Rename HTTPText to TextMap Propagator

### LINKED ISSUE #184: Rename HTTPText to TextMap Propagator.
See the [spec change](https://github.com/open-telemetry/opentelemetry-specification/pull/793) for details.

### PR DESCRIPTION
**Pull request by siyopao**
I guess `examples/tracing-grpc` is constrained by `tracing-opentelemetry` so couldn't be updated atm.

Closes #184.

### CODE REVIEW

File: src/global/propagation.rs [hunk 1]
```diff
@@ -23,20 +23,20 @@ lazy_static::lazy_static! {
 /// // assign it as the global propagator
 /// global::set_http_text_propagator(propagator);
 /// ```
-pub fn set_http_text_propagator<P: api::HttpTextFormat + Send + Sync + 'static>(propagator: P) {
-    let _lock = GLOBAL_HTTP_TEXT_PROPAGATOR
+pub fn set_http_text_propagator<P: api::TextMapFormat + Send + Sync + 'static>(propagator: P) {
```
Comment (jtescher) [inline src/global/propagation.rs:26]: Think we should probably update this method name as well

Comment (siyopao) [reply src/global/propagation.rs]: Whoops. Sorry about that. Fixed.

File: src/global/propagation.rs [hunk 2]
```diff
@@ -21,39 +21,39 @@ lazy_static::lazy_static! {
...
 /// use std::collections::HashMap;
 ///
 /// let example_carrier = HashMap::new();
 ///
 /// // create your http text propagator
 /// let tc_propagator = api::TraceContextPropagator::new();
-/// global::set_http_text_propagator(tc_propagator);
+/// global::set_text_map_propagator(tc_propagator);
 ///
 /// // use the global http text propagator to extract contexts
```
Comment (jtescher) [inline src/global/propagation.rs:48]: So close, sorry missed these the first time around, maybe find and replace for `http text`? 😄

Comment (siyopao) [reply src/global/propagation.rs]: Ugh, can't believe I missed those. Sorry about that. Fixed! And this time I am pretty confident that we got them all 🤞

**Pull request merged**